pkg/canary/scaled_object_reconciler.go

```diff
@@ -47,18 +48,27 @@ func (sor *ScaledObjectReconciler) reconcilePrimaryScaler(cd *flaggerv1.Canary,
 
 	setPrimaryScaledObjectQueries(cd, targetSoClone.Spec.Triggers)
 
+	initializeTargetSoClone(targetSoClone)
+
 	soSpec := keda.ScaledObjectSpec{
 		ScaleTargetRef: &keda.ScaleTarget{
 			Name:                   primaryName,
 			Kind:                   targetSoClone.Spec.ScaleTargetRef.Kind,
 			APIVersion:             targetSoClone.Spec.ScaleTargetRef.APIVersion,
 			EnvSourceContainerName: targetSoClone.Spec.ScaleTargetRef.EnvSourceContainerName,
 		},
-		PollingInterval:  targetSoClone.Spec.PollingInterval,
-		CooldownPeriod:   targetSoClone.Spec.CooldownPeriod,
-		MinReplicaCount:  targetSoClone.Spec.MinReplicaCount,
-		MaxReplicaCount:  targetSoClone.Spec.MaxReplicaCount,
-		Advanced:         targetSoClone.Spec.Advanced,
+		PollingInterval: targetSoClone.Spec.PollingInterval,
+		CooldownPeriod:  targetSoClone.Spec.CooldownPeriod,
+		MinReplicaCount: targetSoClone.Spec.MinReplicaCount,
+		MaxReplicaCount: targetSoClone.Spec.MaxReplicaCount,
+		// Allow the hpa name to be set in the target scaled object, so that it can be used in the primary scaled object
+		Advanced: &keda.AdvancedConfig{
```

Comment:
i think this can still be `Advanced: targetSoClone.Spec.Advanced`, since we've added `Name` to the `HorizontalPodAutoscalerConfig` definition.

Reply:
Ah yes you are correct, I will update. Thanks